atmo/jobs/models.py

```diff
@@ -395,24 +395,26 @@ def sync(self, info=None):
 
         # if the job cluster terminated with error raise the alarm
         if self.status == Cluster.STATUS_TERMINATED_WITH_ERRORS:
-            transaction.on_commit(lambda: SparkJobRunAlert.objects.create(
-                run=self,
-                reason_code=info['state_change_reason_code'],
-                reason_message=info['state_change_reason_message'],
-            ))
+            transaction.on_commit(lambda: self.alert(info))
         self.save()
         return self.status
 
+    def alert(self, info):
+        self.alerts.get_or_create(
+            reason_code=info['state_change_reason_code'],
+            reason_message=info['state_change_reason_message'],
+        )
```

Comment:
Given that this is expected to return only one object at the most, should there be a `unique_together` on `SparkJobRunAlert` for `(run, reason_code, reason_message)`?